snuba/stateful_consumer/states/catching_up.py

```diff
@@ -13,13 +17,29 @@ class CatchingUpState(State[ConsumerStateCompletionEvent, ConsumerStateData]):
     consumption.
     """
 
+    def __init__(
+        self,
+        consumer_builder: ConsumerBuilder
+    ) -> None:
+        super(CatchingUpState, self).__init__()
+        self.__consumer_builder = consumer_builder
+        self.__consumer = None
+
     def signal_shutdown(self) -> None:
-        pass
+        if self.__consumer:
+            self.__consumer.signal_shutdown()
 
     def handle(self, state_data: ConsumerStateData) -> Tuple[ConsumerStateCompletionEvent, ConsumerStateData]:
-        # TODO: Actually consume cdc topic while discarding xids that were
-        # already in the dump
+        assert state_data.snapshot_id is not None
+        assert state_data.transaction_data is not None
```

Comment:
Re: `ConsumerStateData` — more evidence this should be `None` or a value with fields set.